[charts] Fix typo in performance note for CustomLineMarks demo (#20529)
mui/mui-x: docs/data/charts/line-demo/line-demo.md

## CustomLineMarks

Notice that using another shape than "circle" renders a `<path />` instead of the `<circle />` for mark elements.

> This modification implies a small drop of rendering performances (around +50ms to render 1,000 marks).

Reviewer: Small rewording:

```suggestion
This modification results in a small drop in rendering performance (around 50ms to render 1,000 marks).
```
